src/components/NftPreview/NftPreview.tsx

```diff
@@ -38,23 +60,23 @@ function NftPreview({ nft, collectionId, columns }: Props) {
       event.stopPropagation();
       // TODO: Should refactor to utilize navigation context instead of session storage
       if (storage) storage.setItem('prevPage', window.location.pathname);
-      navigateToUrl(`/${username}/${collectionId}/${nft.id}`, event);
+      navigateToUrl(`/${username}/${collection.dbid}/${nft.dbid}`, event);
     },
-    [collectionId, navigateToUrl, nft.id, storage, username]
+    [collection.id, navigateToUrl, nft.dbid, storage, username]
   );
   const isMobile = useIsMobileWindowWidth();
 
   // width for rendering so that we request the apprpriate size image.
   const previewSize = isMobile ? MOBILE_NFT_WIDTH : LAYOUT_DIMENSIONS[columns];
 
   return (
-    <StyledNftPreview key={nft.id} columns={columns}>
+    <StyledNftPreview key={nft.dbid} columns={columns}>
       <StyledLinkWrapper onClick={handleNftClick}>
         <ShimmerProvider>
           {/* // we'll request images at double the size of the element so that it looks sharp on retina */}
-          <NftPreviewAsset nft={nft} size={previewSize * 2} />
+          <NftPreviewAsset nftRef={nft} size={previewSize * 2} />
           <StyledNftFooter>
-            <StyledNftLabel nft={nft} />
+            <StyledNftLabel title={nft.name} collectionName={'THIS IS BUSTED'} />
```

Comment:
@tbezman good catch, this is gonna be broken for the time being but that's cool! will hide it from view